src/v/datalake/coordinator/coordinator.h

```diff
@@ -77,6 +77,7 @@ class coordinator {
     struct last_offsets {
         std::optional<kafka::offset> last_added_offset;
         std::optional<kafka::offset> last_committed_offset;
+        std::optional<model::timestamp> last_added_timestamp;
```

Comment:
Is this the Kafka data batch timestamp? Or something else? If Kafka, we might want to consider using a different timestamp, if this is going to be used for deadline tracking. Users can supply totally bogus timestamps and I can imagine that would mess with scheduling in unpredictable ways

I know we've chatted about other options, one that isn't great but might be good enough for our needs is something like time since translating to the HWM

Reply:
i just shoved whatever was available already in the mux to expedite the mechanical changes. but yeah, right now it's

```c++
batch.header().first_timestamp.value() + record.timestamp_delta()
```

> Users can supply totally bogus timestamps and I can imagine that would mess with scheduling in unpredictable ways

This is intended as feedback for the @mmaslankaprv's throttling controller, but I assume same reasoning applies.

> time since translating to the HWM

sorry, what is "HWM"?

Reply:
>sorry, what is "HWM"?

The high watermark, or I guess in this case I should've said last stable offset / max translatable offset

Reply:
>thought: none of these really accounts for idle topics

Yeah, I guess in my mind, accounting for idle topics can/should be separated from deadline tracking. Truly idle topics won't even be considered by the scheduler (no work to do), but _kind of_ idle topics will be considered and should just be scheduled at a lower priority, no?

If the question is more about how throttling should account for it, I'm thinking could we use a similar idea, where topics that are fully idle (have translated to the max translatable) have 0 lag always? Or is there another consideration I'm missing?

Reply:
ah, oops. yeah, had a couple of translator operations out of sequence in my head. this sounds right to me.

Reply:
done. have a look